homework6/task2_oop2/tests/test_oop2.py

```diff
@@ -0,0 +1,115 @@
+import datetime
+
+import pytest
+from project2_oop2.oop2 import (
+    DeadlineError,
+    Homework,
+    HomeworkResult,
+    HomeworkTypeError,
+    Student,
+    Teacher,
+)
+
+
+def test_teacher_class_attributes():
+    opp_teacher = Teacher("Daniil", "Shadrin")
+
+    assert opp_teacher.first_name == "Daniil"
+    assert opp_teacher.last_name == "Shadrin"
+
+
+def test_student_class_attributes():
+    lazy_student = Student("Roman", "Petrov")
+
+    assert lazy_student.first_name == "Roman"
+    assert lazy_student.last_name == "Petrov"
+
+
+def test_teacher_create_homework():
+    opp_teacher = Teacher("Daniil", "Shadrin")
+    oop_hw = opp_teacher.create_homework("Learn OOP", 1)
+
+    assert oop_hw.text == "Learn OOP"
+    assert oop_hw.deadline == datetime.timedelta(days=1)
+
+
+def test_active_homework_return_homeworkresult_class_object_positive_test():
```

Comment:
You can skip technical details.  Use test to speak in terms of business. `test_active_homework_is_done_by_student`